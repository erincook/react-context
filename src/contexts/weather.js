import React, {useState, createContext} from 'react';
export const WeatherContext = createContext(); 
export const WeatherProvider = props => {
    const [weatherList, setWeatherList] = useState([
        {
            location: "Canton,GA,USA",
            weather: {
                "coord": {
                    "lon": -81.38,
                    "lat": 40.8
                },
                "weather": [
                    {
                        "id": 804,
                        "main": "Clouds",
                        "description": "overcast clouds",
                        "icon": "04d"
                    }
                ],
                "base": "stations",
                "main": {
                    "temp": 278.57,
                    "feels_like": 272.63,
                    "temp_min": 278.15,
                    "temp_max": 279.15,
                    "pressure": 1016,
                    "humidity": 81
                },
                "visibility": 10000,
                "wind": {
                    "speed": 6.2,
                    "deg": 340
                },
                "clouds": {
                    "all": 90
                },
                "dt": 1604066728,
                "sys": {
                    "type": 1,
                    "id": 3573,
                    "country": "US",
                    "sunrise": 1604058850,
                    "sunset": 1604096638
                },
                "timezone": -14400,
                "id": 5149222,
                "name": "Canton",
                "cod": 200
            }
        }
    ]); 
   return (
       <WeatherContext.Provider value={[weatherList, setWeatherList]}>
           {props.children}
       </WeatherContext.Provider>
   )
}