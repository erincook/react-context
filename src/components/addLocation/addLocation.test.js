import React from 'react';
import ReactDOM from 'react-dom'; 
import { act } from 'react-dom/test-utils';
import {WeatherProvider} from '../../contexts/weather';
import AddLocation from "./addLocation";
import mockJson from "../../mockjson.js";  
import renderer from 'react-test-renderer'; // For snapshot testing. 

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve(mockJson),
  })
);

let container; 
beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
});
  
afterEach(() => {
    document.body.removeChild(container);
    container = null;
});

it("renders without crashing", () => {
    ReactDOM.render(<WeatherProvider><AddLocation/></WeatherProvider>, container); 
});

it("should match snapshot", () => {
    const tree = renderer.create(<WeatherProvider><AddLocation/></WeatherProvider>).toJSON(); 
    expect(tree).toMatchSnapshot(); 
}); 

it("should call fetch once after button is clicked", () => {
    act(()=>{ ReactDOM.render(<WeatherProvider><AddLocation/></WeatherProvider>, container); }); 
    const input = container.querySelector("input"); 
    const button = container.querySelector("button"); 
    act(()=> {
        input.value = "Canton,GA,USA"; 
        button.click(); 
    }); 
    expect(fetch).toHaveBeenCalledTimes(1);
}); 
