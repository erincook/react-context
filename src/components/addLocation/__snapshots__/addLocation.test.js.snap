// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="add-location"
>
  <input
    name="location-input"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Add Location
  </button>
  <div>
    
  </div>
</div>
`;
